src/hotspot/share/opto/compile.cpp

```diff
@@ -2104,9 +2104,16 @@ void Compile::Optimize() {
       ResourceMark rm;
       PhaseRenumberLive prl = PhaseRenumberLive(initial_gvn(), for_igvn(), &new_worklist);
     }
+    Unique_Node_List* save_for_igvn = for_igvn();
     set_for_igvn(&new_worklist);
     igvn = PhaseIterGVN(initial_gvn());
     igvn.optimize();
+
+    // Restore the original for_igvn pointer for subsequent uses.
+    // Clear the for_igvn list because it may have irrelevant nodes
+    // from the previous PhaseRenumberLive run.
+    save_for_igvn->clear();
```

Comment:
Since `PhaseRenumberLive::PhaseRenumberLive` moves nodes from `for_igvn()` to `new_worklist`, does it make more sense to drain original `for_igvn()` worklist there?

Reply:
+1

Reply:
Thank you for the suggestion. I have updated the PR.

Reply:
LGTM